pkg/providers/oidc/verifier.go

```diff
@@ -107,3 +120,30 @@ func (v *idTokenVerifier) interfaceSliceToString(slice interface{}) []string {
 	}
 	return strings
 }
+
+func (v *idTokenVerifier) verifyEmail(token *oidc.IDToken) (bool, error) {
+
+	if v.verificationOptions.AllowUnverifiedEmail {
+		return true, nil
+	}
```

Comment:
shouldn't this be part of the `if claims.Verified != nil && !*claims.Verified` in line 144 as before because moving it before the `token.Claims(&claims)` logic and the replacement of empty `claims.Email` with the `claims.Subject` could lead to a breaking change?